geps/gep-2643/index.md

```diff
@@ -0,0 +1,439 @@
+  // match with the criteria above, then the TLSRoute is not accepted for that 
+  // Listener. If the TLSRoute does not match any Listener on the parent, the 
+  // implementation must raise an 'Accepted' Condition with a status of
+  // `False` in the corresponding RouteParentStatus.
+  // 
+  // A Listener MUST be of type TLS when a TLSRoute attaches to it. The 
+  // implementation MUST raise an 'Accepted' Condition with a status of
+  // `False` in the corresponding RouteParentStatus with the reason 
+  // of "UnsupportedValue" in case a Listener of the wrong type is used.
+  // Core: A TLS listener CAN have mode Passthrough
+  // Extended: A TLS listener CAN have mode Terminate 
+  // </gateway:util:excludeFromCRD>
+  // +required
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  Hostnames []Hostname `json:"hostnames,omitempty"`
+  // Rules is a list of TLS matchers and actions.
+  //
+  // +required
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=1
+  Rules []TLSRouteRule `json:"rules,omitempty"`
+}
+
+type TLSRouteRule struct {
+  // Name is the name of the route rule. This name MUST be unique within a Route if it is set.
+  //
+  // +optional
+  Name *SectionName `json:"name,omitempty"`
+  // BackendRefs (same as other BackendRef here)
+  // +required
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  BackendRefs []BackendRef `json:"backendRefs,omitempty"`
+}
+```
+
+## Request flow
+
+Following are some of the requests flow covered by TLSRoute
+
+### TLSRoute + TLS Passthrough
+In this workflow, the TLS traffic will be matched against the `SNI attribute` of 
+the request and then directed to the backends.
+
+This workflow MUST be supported on Core support level.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway-tlsroute
+spec:
+  gatewayClassName: "my-class"
+  listeners:
+  - name: somelistener
+    port: 443
+    protocol: TLS
+    hostname: "*.example.com"
+    allowedRoutes:
+      namespaces:
+        from: Same
+      kinds:
+      - kind: TLSRoute
+    tls:
+      mode: Passthrough
+---
+apiVersion: gateway.networking.k8s.io/v1alpha3
+kind: TLSRoute
+metadata:
+  name: my-tls-route
+spec:
+  parentRefs:
+  - name: gateway-tlsroute
+  hostnames:
+  - foo.example.com
+  rules:
+  - backendRefs:
+    - name: tls-backend
+      port: 443
+```
+
+A typical [north/south](https://gateway-api.sigs.k8s.io/concepts/glossary/#northsouth-traffic)
+API request flow for a gateway implemented using a `TLSRoute` is:
+* A client makes a request to https://foo.example.com.
+* DNS resolves the name to a `Gateway` address.
+* The reverse proxy receives the request on a `Listener` and uses the
+[Server Name Indication](https://datatracker.ietf.org/doc/html/rfc6066#section-3)
+attribute to match an `TLSRoute`.
+* The reverse proxy passes through the request directly to one or more objects,
```

Comment:
What would an example of passing to multiple objects?

Reply:
I see, I have fixed it here